doc/examples/boilerplate.py

```diff
@@ -0,0 +1,135 @@
+"""
+A docstring
+
+This can double as a helpful message which explains how the program is run.
+"""
+
+from __future__ import absolute_import, division, print_function
+
+import logging
+import sys
+from typing import List  # noqa: F401  Flake8 doesn't recognise Python 2 typing.
```

Comment:
Are we ready to start using typing?

Reply:
I think it often does exist on our installs but only as a chained dependency. In python3 it's part of the standard library, so I think for now the better pattern is:
```
try:
    from typing import List
except ImportError:
    pass
```
Flake8 and other tools will accept this as an "imported symbol" and stop the type warnings

Reply:
Great, thanks @ndevenish, I'll do that.

@graeme-winter, I think it's very helpful if we can try to use it, since it's not often clear to 
someone who is new to a particular corner of the code base which in-house special objects are being passed around.  With Nick's small addition, it's almost painless and doesn't break anything.